fetch.js

```diff
@@ -526,13 +526,13 @@ export function fetch(input, init) {
 
     xhr.onerror = function() {
       setTimeout(function() {
-        reject(new TypeError('Network request failed'))
+        reject(new TypeError('Network request failed: ' + request.url))
       }, 0)
     }
 
     xhr.ontimeout = function() {
       setTimeout(function() {
-        reject(new TypeError('Network request failed'))
+        reject(new TypeError('Network request failed: ' + request.url))
```

Comment:
While this is a noble change to improve the debugging quality, such error would be incompatible with a TypeError thrown by `window.fetch`:

```js
fetch('http://foo.bar')
// Uncaught (in promise) TypeError: Failed to fetch
//    at <anonymous>:1:1
```

This repository is a polyfill for `window.fetch`, which means that it treats `window.fetch` as the source of truth and a specification to look up to.

Reply:
That makes sense, thank you very much for the detailed answer.